### ISSUE TITLE
fission fn test returns unhelpful 502 on failed specializations

### ISSUE THREAD

**Issue by erwinvaneyk** [fission/fission#908]
Triggering functions that failed to specialise results in the 502 error on invoking. This is not very helpful to the user as they will have to dig through the logs to uncover the cause of the error. I would expect that, since the executor knows that building has failed, it should return the error message of environment pod the instead of 502.

### Example
Executor logs:
```
2018/09/28 12:37:28 [inputs] Checking for cached function service
2018/09/28 12:37:28 [inputs] No cached function service found, creating one
2018/09/28 12:37:28 [inputs] getting environment for function
2018/09/28 12:37:28 [&ObjectMeta{Name:inputs,GenerateName:,Namespace:default,SelfLink:/apis/fission.io/v1/namespaces/default/functions/inputs,UID:3faf9893-c31b-11e8-9a55-42010a800070,ResourceVersion:29700741,Generation:1,CreationTimestamp:2018-09-28 12:37:23 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,}] getting env
2018/09/28 12:37:28 [inputs] getting function service from pool
2018/09/28 12:37:28 [inputs] Choosing pod from pool
2018/09/28 12:37:28 [map[unmanaged:true functionName:inputs functionUid:3faf9893-c31b-11e8-9a55-42010a800070 executorInstanceId:exdoo13v]] found 1 ready pods of 1 total
2018/09/28 12:37:28 Chosen pod: workflow-3b1f7faa-c31b-11e8-9a55-42010a800070-3z56lxrr-99fjvvq8 (in 5.081866ms)
2018/09/28 12:37:28 [inputs] calling fetcher to copy function with fetcher url: http://10.20.0.150:8000/
2018/09/28 12:37:28 [inputs] specializing pod
2018/09/28 12:37:28 specialize url: http://10.20.0.150:8888/v2/specialize
2018/09/28 12:37:28 Failed to specialize pod: Internal error - failed to specialize package: failed to parse bytes into WorkflowSpec: json: cannot unmarshal string into Go value of type map[string]*json.RawMessage
2018/09/28 12:37:28 [inputs] Error creating service for function: Internal error - failed to specialize package: failed to parse bytes into WorkflowSpec: json: cannot unmarshal string into Go value of type map[string]*json.RawMessage
2018/09/28 12:37:28 Error: 500: [inputs] Error creating service for function: Internal error - failed to specialize package: failed to parse bytes into WorkflowSpec: json: cannot unmarshal string into Go value of type map[string]*json.RawMessage
10.20.0.148 - - [28/Sep/2018:12:37:28 +0000] "POST /v2/getServiceForFunction HTTP/1.1" 500 214
2018/09/28 12:37:28 Error in pod 'workflow-3b1f7faa-c31b-11e8-9a55-42010a800070-3z56lxrr-99fjvvq8', scheduling cleanup
time="2018-09-28T12:37:34Z" level=info msg="List watch for package reported a new package addition: inputs-wf-yaml-6uxr.default" 
2018/09/28 12:37:34 SA : fission-fetcher.fission-function already present in rolebinding : package-getter-binding.default, so nothing to add
time="2018-09-28T12:37:34Z" level=info msg="Successfully set up rolebinding for fetcher SA: fission-fetcher.fission-function, in packages's ns : default, for pkg : inputs-wf-yaml-6uxr" 
2018/09/28 12:37:34 SA : fission-fetcher.fission-function already present in rolebinding : secret-configmap-getter-binding.default, so nothing to add
time="2018-09-28T12:37:34Z" level=info msg="Successfully set up rolebinding for fetcher SA: fission-fetcher.fission-function, in func's ns for func : default/inputs" 
```

router logs
```
2018/09/28 12:37:28 Calling getServiceForFunction for function: inputs
2018/09/28 12:37:28 http: proxy error: Internal error - [inputs] Error creating service for function: Internal error - failed to specialize package: failed to parse bytes into WorkflowSpec: json: cannot unmarshal string into Go value of type map[string]*json.RawMessage
```

user terminal
```
$ fission fn create --name inputs --env workflow --src test/e2e/tests/../../../examples/misc/inputs.wf.yaml
function 'inputs' created
$ +fission fn test --name inputs -b foobar -H 'HEADER_KEY: HEADER_VAL' --method PUT
Error calling function inputs: 502
```

TL;DR even though the executor and router are both aware of the particular error message of the environment pod, this is not propagated to the user.

Source: https://api.travis-ci.org/v3/job/434588689/log.txt
